### ISSUE TITLE
Line2d's antialiased is not working correctly in Gles3d

### ISSUE THREAD

**Issue by gamergamesz** [godotengine/godot#38872]
Godot_v3.2.2-beta2_win64

windows 10

Godot_v3.2.2-beta2_win64

Gles3
![image](https://user-images.githubusercontent.com/13883739/82368246-bdb4b380-9a32-11ea-805c-9769dfb2e152.png)

Gles2
![image](https://user-images.githubusercontent.com/13883739/82368303-d6bd6480-9a32-11ea-8e5d-8ac8e2294dfe.png)

Querry - Is Gles2 deprecating?

**Comment by gamergamesz** [comment]
Currently, there is one more issue I am facing but I will report it after confirmation.
The issue is that while using shaders with Gles2, my android game runs ok with the quick install but after restarting it the game crashes! I have not encountered the crash in Gles 3 yet

**Comment by EricEzaM** [comment]
Try turning HDR off if you have it on.

Also gles 3 will be replaced by Vulkan in 4.0,and gles 2 will stay.

**Comment by gamergamesz** [comment]
https://mobile.twitter.com/reduzio/status/1254446329801191424

**Comment by lawnjelly** [comment]
I did do a little test with this a couple of days ago, didn't come up with any firm conclusions. I couldn't seem to get anti-aliasing in either GLES2 or GLES3 on my machine, while stepping through the code did show it was drawing the relevant lines.

There may be some subtle state bug, I'm not really that familiar with the underlying state logic for lines (I tried ensuring blending was on, this didn't fix it). The degree of anti-aliasing that is done may also just be dependent on the driver and hardware. There are some glHints available for the niceness of the lines, haven't tried those.

**Comment by ttencate** [comment]
I'm getting antialiasing in GLES2 on Godot 3.2.2-rc2 when running natively on Linux, but in the HTML5 export, the antialiasing is gone.

Repro project is attached, but it's trivial. Just create a new project using GLES2, add a `Line2D` node, draw some vertices, and check its Antialiased checkbox. Then export to HTML5 and open in browser.
[line2d_antialiasing_38872.zip](https://github.com/godotengine/godot/files/4812785/line2d_antialiasing_38872.zip)

**Comment by Calinou** [comment]
@ttencate I think WebGL 1.0 doesn't mandate support for line smoothing (or even supports it in the first place). We have to draw lines using rectangles and use additional vertices or MSAA to solve this.

**Comment by ttencate** [comment]
That's fair, but maybe it should be mentioned in the docs for the `antialias` property then.

Doesn't Godot tesselate the line anyway, in order to support texuring, end caps and so on?

**Comment by Calinou** [comment]
> Doesn't Godot tesselate the line anyway, in order to support texuring, end caps and so on?

You're right, it actually does this as long as the line width is greater than `1`.

**Comment by lawnjelly** [comment]
> Doesn't Godot tesselate the line anyway, in order to support texuring, end caps and so on?

With line width greater than 1, it draws a central rect then draws lines around the outside. So the anti-aliasing is still an issue in either case.

**Comment by Xrayez** [comment]
I've actually managed to draw antialiased geometry in GLES3 (the issue doesn't seem to be isolated to line drawing as seen in #43618), but it only works while drawing via `EditorPlugin.forward_canvas_draw_over_viewport()` method, I have no clue why.
